maven-core/src/main/java/org/apache/maven/lifecycle/DefaultLifecycles.java

```diff
@@ -45,7 +45,7 @@
 @Singleton
 public class DefaultLifecycles
 {
-    public static final String[] STANDARD_LIFECYCLES = { "default", "clean", "site", "wrapper" };
+    public static final String[] STANDARD_LIFECYCLES = { "clean", "default", "site", "wrapper" };
```

Comment:
Does the order make change make any difference since the next hunk says so?!

Reply:
Yes, the change is required.  

The current code (without this PR) provides two different ordering for the lifecycles, one is given by `DefaultLifecycles#getLifeCycles()` which sorts the lifecycles according to the `STANDARD_LIFECYCLES` array.  The other one comes `DefaultLifecyclePluginAnalyzer#getOrderedLifecycles()` which sorts the lifecycles alphabetically.
My analysis is that the first order isn't really used, while the second one is actually while computing the goals that will be invoked from the phases given on the command line.
I thus changed the first order to actually return the same thing that the second one and not rely on alphabetical ordering which imho does not make sense semantically : the `clean` phase should be executed before the `default` and the `site` after the `default`, and the alphabetical order is not really relevant.

If it's problematic, I'll revert that part and will provide another sort just for the use case.

Reply:
I see, your explanation makes sense and simplifies the process.